// Copyright Â© 2025 InfraMatrix. All Rights Reserved.

// SPDX-License-Identifier: BSD-3-Clause

syntax = "proto3";

package compute;

service vmm {

  rpc GetVMS(GetVMSRequest) returns (GetVMSResponse);
  rpc CreateVM(CreateVMRequest) returns (CreateVMResponse);
  rpc DeleteVM(DeleteVMRequest) returns (DeleteVMResponse);
  rpc StartVM(StartVMRequest) returns (StartVMResponse);
  rpc ShutdownVM(ShutdownVMRequest) returns (ShutdownVMResponse);
  rpc ResumeVM(ResumeVMRequest) returns (ResumeVMResponse);
  rpc StopVM(StopVMRequest) returns (StopVMResponse);
  rpc GetVMStatus(GetVMStatusRequest) returns (GetVMStatusResponse);
  rpc GetVMIP(GetVMIPRequest) returns (GetVMIPResponse);
  rpc StartPTYConnection (StartPTYConnectionRequest) returns (StartPTYConnectionResponse);

}

message GetVMSRequest {
  int64 status = 1;
}
message GetVMSResponse {
    repeated string vm_names = 1;
}

message CreateVMRequest {}
message CreateVMResponse {
  string vm_name = 1;
}

message DeleteVMRequest {
  int64 vm_number = 1;
}
message DeleteVMResponse {
  string vm_name = 1;
}

message StartVMRequest {
  int64 vm_number = 1;
}
message StartVMResponse {
  string vm_name = 1;
}

message ShutdownVMRequest {
  int64 vm_number = 1;
}
message ShutdownVMResponse {
  string vm_name = 1;
}

message ResumeVMRequest {
  int64 vm_number = 1;
}
message ResumeVMResponse {
  string vm_name = 1;
}

message StopVMRequest {
  int64 vm_number = 1;
}
message StopVMResponse {
  string vm_name = 1;
}

message GetVMStatusRequest {
  int64 vm_number = 1;
}
message GetVMStatusResponse {
  string vm_status = 1;
}

message GetVMIPRequest {
  int64 vm_number = 1;
}
message GetVMIPResponse {
  string vm_ip_addr = 1;
  string vm_ip_port = 2;

}

message StartPTYConnectionRequest {
  int64 vm_number = 1;
}
message StartPTYConnectionResponse {
  int64 vm_number = 1;
}
